go.mod

```diff
@@ -1,13 +1,13 @@
 module github.com/open-telemetry/opentelemetry-ebpf-profiler
 
-go 1.22.2
+go 1.23.1
```

Comment:
Should we support more than just the latest Go version?
The Go SDK uses the same support policy as for the Go language: https://go.dev/doc/devel/release

Which means supporting 1.22 and 1.23.

Reply:
:+1: Makes sense given that the code is used as library by different projects.

Reply:
Reverted the Go version bump and updated the PR title and description.